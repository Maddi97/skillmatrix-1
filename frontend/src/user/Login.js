import RestPoints from './Init';
import RestCom from './Rest';

export default async function (username, password)
{
    // map data to get the right data structure for posting to server
    const user = {
        username,
        password
    }

    const Rest = new RestCom(RestPoints.login, user);

    //post and get response from server
    const response = await Rest.post();

    return errorCodesToErrorMsg(response)
}


//translate error codes to error Msg and return a new map that looks like:
//response = {user: {username: String}, errorMsg: String, success: Bool}
const errorCodesToErrorMsg = (apiResponse) => {

    const response = {
        user: "",
        errorMsg: "",
        success: true
    }

    //there was an Error connecting:
    if (apiResponse === undefined){
        response.errorMsg = "An error occured while connecting to the database. Check your Internet connection.";
        response.success = false;
        return response;
    }

    const data = apiResponse.data;
    
    //looks like {user: {username: String, ....}}
    const user = data.user;

    //holds the error code
    const error = data.error;

    switch (error){
        case 0:
            response.user = user;
            response.success = true;
            break;
        case 1:
            response.errorMsg = "Wrong login credentials.";
            response.success = false;
            break;
        case 2:
            response.errorMsg = "Please fill in all form fields."
            response.success = false;
            break;
        case 3:
            response.errorMsg = "Server Timeout. Please try again.";
            response.success = false;
            break;
        default:
            response.errorMsg = "An unknown Error occurred. Please try again.";
            response.success = false;
            break;
    }
    return response;
}