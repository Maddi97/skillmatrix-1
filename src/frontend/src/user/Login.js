import RestPoints from './Init';
import RestCom from './Rest';

export default async function (username, password) {
  // map data to get the right data structure for posting to server
  const loginCredentials = {
    username,
    password,
  };

  let response = {};

  try {
    const Rest = new RestCom(RestPoints.login, loginCredentials);
    // post and get response from server
    const { data } = await Rest.post();
    const { user } = data;
    response = {
      user,
      success: true,
    };
  } catch (e) {
    const errorMsg = e.message;
    response = {
      errorMsg,
      success: false,
    };
  }

  return response;
}
