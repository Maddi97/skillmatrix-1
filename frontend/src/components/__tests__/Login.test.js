import React from 'react';
import { render,shallow, mount } from 'enzyme';
import Login from '../Login'


test('render Login-Component', () => {
    

    //mockt login aus index.js
    const wrapper = render(<Login/>);
    expect(wrapper).toMatchSnapshot();

});