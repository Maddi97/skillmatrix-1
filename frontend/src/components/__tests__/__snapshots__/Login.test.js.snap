// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pure function returns the same output for the same input 1`] = `
<main
  class="SignIn-main-1"
>
  <div
    class="MuiPaper-root-6 MuiPaper-elevation2-10 MuiPaper-rounded-7 SignIn-paper-2"
  >
    <div
      class="MuiAvatar-root-33 MuiAvatar-colorDefault-34 SignIn-avatar-3"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root-36"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <g
          fill="none"
        >
          <path
            d="M0 0h24v24H0V0z"
          />
          <path
            d="M0 0h24v24H0V0z"
            opacity=".87"
          />
        </g>
        <path
          d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
        />
      </svg>
    </div>
    <h1
      class="MuiTypography-root-45 MuiTypography-h4-60"
    >
      Skill Matrix
    </h1>
    <form
      class="SignIn-form-4"
    >
      <div
        class="MuiFormControl-root-80 MuiFormControl-marginNormal-81 MuiFormControl-fullWidth-83"
      >
        <label
          class="MuiFormLabel-root-91 MuiFormLabel-required-96 MuiInputLabel-root-84 MuiInputLabel-formControl-85 MuiInputLabel-animated-88"
          data-shrink="false"
          for="email"
        >
          Email Address
          <span
            class="MuiFormLabel-asterisk-97"
          >
             *
          </span>
        </label>
        <div
          class="MuiInputBase-root-111 MuiInput-root-98 MuiInput-underline-102 MuiInputBase-formControl-112 MuiInput-formControl-99"
        >
          <input
            aria-invalid="false"
            autocomplete="email"
            autofocus=""
            class="MuiInputBase-input-121 MuiInput-input-106"
            id="email"
            name="email"
            required=""
            type="text"
          />
        </div>
      </div>
      <div
        class="MuiFormControl-root-80 MuiFormControl-marginNormal-81 MuiFormControl-fullWidth-83"
      >
        <label
          class="MuiFormLabel-root-91 MuiFormLabel-required-96 MuiInputLabel-root-84 MuiInputLabel-formControl-85 MuiInputLabel-animated-88"
          data-shrink="false"
          for="password"
        >
          Password
          <span
            class="MuiFormLabel-asterisk-97"
          >
             *
          </span>
        </label>
        <div
          class="MuiInputBase-root-111 MuiInput-root-98 MuiInput-underline-102 MuiInputBase-formControl-112 MuiInput-formControl-99"
        >
          <input
            aria-invalid="false"
            autocomplete="current-password"
            class="MuiInputBase-input-121 MuiInput-input-106 MuiInputBase-inputType-124 MuiInput-inputType-109"
            id="password"
            name="password"
            required=""
            type="password"
          />
        </div>
      </div>
      <button
        class="MuiButtonBase-root-154 MuiButton-root-128 MuiButton-contained-139 MuiButton-containedPrimary-140 MuiButton-raised-142 MuiButton-raisedPrimary-143 MuiButton-fullWidth-153 SignIn-submit-5"
        tabindex="0"
        type="submit"
      >
        <span
          class="MuiButton-label-129"
        >
          Sign in
        </span>
      </button>
    </form>
  </div>
</main>
`;

exports[`pure function returns the same output for the same input 2`] = `
<form
  class="SignIn-form-4"
>
  <div
    class="MuiFormControl-root-80 MuiFormControl-marginNormal-81 MuiFormControl-fullWidth-83"
  >
    <label
      class="MuiFormLabel-root-91 MuiFormLabel-required-96 MuiInputLabel-root-84 MuiInputLabel-formControl-85 MuiInputLabel-animated-88"
      data-shrink="false"
      for="email"
    >
      Email Address
      <span
        class="MuiFormLabel-asterisk-97"
      >
         *
      </span>
    </label>
    <div
      class="MuiInputBase-root-111 MuiInput-root-98 MuiInput-underline-102 MuiInputBase-formControl-112 MuiInput-formControl-99"
    >
      <input
        aria-invalid="false"
        autocomplete="email"
        autofocus=""
        class="MuiInputBase-input-121 MuiInput-input-106"
        id="email"
        name="email"
        required=""
        type="text"
      />
    </div>
  </div>
  <div
    class="MuiFormControl-root-80 MuiFormControl-marginNormal-81 MuiFormControl-fullWidth-83"
  >
    <label
      class="MuiFormLabel-root-91 MuiFormLabel-required-96 MuiInputLabel-root-84 MuiInputLabel-formControl-85 MuiInputLabel-animated-88"
      data-shrink="false"
      for="password"
    >
      Password
      <span
        class="MuiFormLabel-asterisk-97"
      >
         *
      </span>
    </label>
    <div
      class="MuiInputBase-root-111 MuiInput-root-98 MuiInput-underline-102 MuiInputBase-formControl-112 MuiInput-formControl-99"
    >
      <input
        aria-invalid="false"
        autocomplete="current-password"
        class="MuiInputBase-input-121 MuiInput-input-106 MuiInputBase-inputType-124 MuiInput-inputType-109"
        id="password"
        name="password"
        required=""
        type="password"
      />
    </div>
  </div>
  <button
    class="MuiButtonBase-root-154 MuiButton-root-128 MuiButton-contained-139 MuiButton-containedPrimary-140 MuiButton-raised-142 MuiButton-raisedPrimary-143 MuiButton-fullWidth-153 SignIn-submit-5"
    tabindex="0"
    type="submit"
  >
    <span
      class="MuiButton-label-129"
    >
      Sign in
    </span>
  </button>
</form>
`;

exports[`render Login-Component 1`] = `
<main
  class="SignIn-main-1"
>
  <div
    class="MuiPaper-root-6 MuiPaper-elevation2-10 MuiPaper-rounded-7 SignIn-paper-2"
  >
    <div
      class="MuiAvatar-root-33 MuiAvatar-colorDefault-34 SignIn-avatar-3"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root-36"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <g
          fill="none"
        >
          <path
            d="M0 0h24v24H0V0z"
          />
          <path
            d="M0 0h24v24H0V0z"
            opacity=".87"
          />
        </g>
        <path
          d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
        />
      </svg>
    </div>
    <h1
      class="MuiTypography-root-45 MuiTypography-h4-60"
    >
      Skill Matrix
    </h1>
    <form
      class="SignIn-form-4"
    >
      <div
        class="MuiFormControl-root-80 MuiFormControl-marginNormal-81 MuiFormControl-fullWidth-83"
      >
        <label
          class="MuiFormLabel-root-91 MuiFormLabel-required-96 MuiInputLabel-root-84 MuiInputLabel-formControl-85 MuiInputLabel-animated-88"
          data-shrink="false"
          for="email"
        >
          Email Address
          <span
            class="MuiFormLabel-asterisk-97"
          >
             *
          </span>
        </label>
        <div
          class="MuiInputBase-root-111 MuiInput-root-98 MuiInput-underline-102 MuiInputBase-formControl-112 MuiInput-formControl-99"
        >
          <input
            aria-invalid="false"
            autocomplete="email"
            autofocus=""
            class="MuiInputBase-input-121 MuiInput-input-106"
            id="email"
            name="email"
            required=""
            type="text"
          />
        </div>
      </div>
      <div
        class="MuiFormControl-root-80 MuiFormControl-marginNormal-81 MuiFormControl-fullWidth-83"
      >
        <label
          class="MuiFormLabel-root-91 MuiFormLabel-required-96 MuiInputLabel-root-84 MuiInputLabel-formControl-85 MuiInputLabel-animated-88"
          data-shrink="false"
          for="password"
        >
          Password
          <span
            class="MuiFormLabel-asterisk-97"
          >
             *
          </span>
        </label>
        <div
          class="MuiInputBase-root-111 MuiInput-root-98 MuiInput-underline-102 MuiInputBase-formControl-112 MuiInput-formControl-99"
        >
          <input
            aria-invalid="false"
            autocomplete="current-password"
            class="MuiInputBase-input-121 MuiInput-input-106 MuiInputBase-inputType-124 MuiInput-inputType-109"
            id="password"
            name="password"
            required=""
            type="password"
          />
        </div>
      </div>
      <button
        class="MuiButtonBase-root-154 MuiButton-root-128 MuiButton-contained-139 MuiButton-containedPrimary-140 MuiButton-raised-142 MuiButton-raisedPrimary-143 MuiButton-fullWidth-153 SignIn-submit-5"
        tabindex="0"
        type="submit"
      >
        <span
          class="MuiButton-label-129"
        >
          Sign in
        </span>
      </button>
    </form>
  </div>
</main>
`;
